### ISSUE TITLE
[Feature] add `requires_grad` key in config

### ISSUE THREAD

**Issue by cxiang26** [open-mmlab/mmengine#750]
### What is the feature?

add `requires_grad` key in `paramwise_cfg` to detach any part of any model flexibly
```python
optim_wrapper = dict(
    type="OptimWrapper",
    optimizer=dict(type="AdamW", lr=lr, weight_decay=0.01),
    paramwise_cfg=dict(
        custom_keys={
            "backbone": dict(requires_grad=False), # lr_mult=0,
            "neck.conv0":dict(requiers_grad=False),
        }
    ),
)
```
1. it is flexible to detach any part of our model, e.g. backbone, neck, neck.xx, ...
2. more memory save than using lr_mult=0
3. it's friendly to all models without standard mmlab-style implementation

**Comment by zhouzaida** [comment]
Hi @cxiang26 , it is a useful feature, and do you have time to support it?

**Comment by cxiang26** [comment]
ok, I will try it this weekend @zhouzaida
